projects/angular/src/data/datagrid/datagrid-virtual-scroll.direcive.ts

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (c) 2016-2024 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Directionality } from '@angular/cdk/bidi';
+import { coerceNumberProperty } from '@angular/cdk/coercion';
+import { _RecycleViewRepeaterStrategy, ListRange } from '@angular/cdk/collections';
+import {
+  CdkFixedSizeVirtualScroll,
+  CdkVirtualForOf,
+  CdkVirtualForOfContext,
+  CdkVirtualScrollableElement,
+  CdkVirtualScrollViewport,
+  FixedSizeVirtualScrollStrategy,
+  ScrollDispatcher,
+  ViewportRuler,
+} from '@angular/cdk/scrolling';
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Directive,
+  DoCheck,
+  ElementRef,
+  EmbeddedViewRef,
+  EnvironmentInjector,
+  EventEmitter,
+  Input,
+  IterableDiffers,
+  NgZone,
+  OnDestroy,
+  Output,
+  SkipSelf,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { ClrDatagrid } from './datagrid';
+import { Items } from './providers/items';
+
+type CdkVirtualForInputKey =
+  | 'cdkVirtualForOf'
+  | 'cdkVirtualForTrackBy'
+  | 'cdkVirtualForTemplate'
+  | 'cdkVirtualForTemplateCacheSize';
+
+type CdkVirtualForInputs<T> = Partial<Pick<CdkVirtualForOf<T>, CdkVirtualForInputKey>>;
+
+type CdkFixedSizeVirtualScrollInputs = Pick<CdkFixedSizeVirtualScroll, 'itemSize' | 'minBufferPx' | 'maxBufferPx'>;
+
+const defaultCdkFixedSizeVirtualScrollInputs: CdkFixedSizeVirtualScrollInputs = {
+  itemSize: 32,
+  minBufferPx: 200,
+  maxBufferPx: 400,
+};
+
+@Directive({
+  selector: '[customClrVirtualRows][customClrVirtualRowsOf]',
+  providers: [Items],
+})
+export class CustomClrVirtualRowsDirective<T> implements AfterViewInit, DoCheck, OnDestroy {
+  @Output() renderedRangeChange = new EventEmitter<ListRange>();
+
+  private _cdkFixedSizeVirtualScrollInputs = { ...defaultCdkFixedSizeVirtualScrollInputs };
+
+  private readonly datagridElementRef: ElementRef<HTMLElement>;
+
+  private gridRoleElement: HTMLElement | null | undefined;
+  private readonly virtualScrollStrategy: FixedSizeVirtualScrollStrategy;
+  private virtualScrollViewport: CdkVirtualScrollViewport;
+  private cdkVirtualFor: CdkVirtualForOf<T>;
+  private subscriptions: Subscription[] = [];
+  private mutationChanges: MutationObserver = new MutationObserver((mutations: MutationRecord[]) => {
+    mutations.forEach((mutation: MutationRecord) => {
+      // it is possible this to be called twice because the old class is removed and the new added
+      if ((mutation.target as HTMLElement).classList.contains('datagrid-compact')) {
```

Comment:
Add condition to check if the value is already the same to skip the itemSize setter and updating the virtual scroll values constantly.
```suggestion
      if ((mutation.target as HTMLElement).classList.contains('datagrid-compact')  && this.itemSize > 24) {
```

Reply:
Corrected in commit 8f155e7.